lib/Image/Operation/Resize.php

```diff
@@ -17,27 +17,51 @@
  */
 class Resize extends ImageOperation {
 
-   private $w, $h, $crop;
+	private $parameters_or_w, $h, $crop;
 
 	/**
-	 * @param int    $w    width of new image
-	 * @param int    $h    height of new image
-	 * @param string $crop cropping method, one of: 'default', 'center', 'top', 'bottom', 'left', 'right', 'top-center', 'bottom-center'.
+	 * @param array|int $parameters_or_w    Array of all information like width, height or cropping.
+	 *                                      Width of new image if integer.
+	 * @param int       $h                  height of new image.
+	 * @param string    $crop               cropping method, one of: 'default', 'center', 'top', 'bottom',
+	 *                                      'left', 'right', 'top-center', 'bottom-center'.
 	 */
-	public function __construct( $w, $h, $crop ) {
-		$this->w = $w;
-		$this->h = $h;
-		// Sanitize crop position
-		$allowed_crop_positions = array('default', 'center', 'top', 'bottom', 'left', 'right', 'top-center', 'bottom-center');
-		if ( $crop !== false && !in_array($crop, $allowed_crop_positions) ) {
-			$crop = $allowed_crop_positions[0];
+	public function __construct( $parameters_or_w, $h, $crop ) {
+
+		$allowed_crop_positions = array( 'default', 'center', 'top', 'bottom', 'left', 'right', 'top-center', 'bottom-center' );
+
+		if ( is_array($parameters_or_w) ) {
+			if ( isset($parameters_or_w['width']) ) {
```

Comment:
Would it maybe help to work with `wp_parse_args()` here instead of a list of if-statements?

```php
$args = wp_parse_args( $parameters_or_w, array(
    'width' => 0,
	'height' => 0,
	// …
) );

$this->w = $args['width'];
$this->h = $args['height'];
// …
```

Reply:
It would also be great to have support for using `w` and `h` instead of `width` and `height`, see https://github.com/timber/timber/issues/857#issuecomment-366797326.

Reply:
`wp_parse_args` is used now - it was really a better solution. Thanks @gchtr . I also changed the variables to `w` and `h`.

Reply:
Oh, I’m sorry! I read my comment again and it’s not what I wanted to say. Maybe I was too tired.

> It would also be great to have support for using `w` and `h` instead of `width` and `height`

What I meant to say was: In addition to the `width` and `height` parameters, it would also be great to have `w` and `h` shorthands.

So I would keep the variable names `width` and `height` – they are surely more readable, but add something like:

```php
$shorthands = array(
    'w' => 'width',
    'h' => 'height',
);

foreach ( $shorthands as $shorthand => $name ) {
    if ( isset( $args[ $shorthand ] ) ) {
        $this->$name = $args[ $shorthand ];
    }
}
```

What do you think? Does that make sense?

Reply:
I thought about it a bit and I would go the other way around. 

```
$shorthands = array(
	'w' => 'width',
	'h' => 'height',
	'crop' => 'c',
);

foreach ( $shorthands as $shorthand => $name ) {
	if ( isset( $args[ $shorthand ] ) ) {
		$this->$shorthand = $args[ $shorthand ];
	} elseif ( isset( $args[ $name ] ) ) {
		$this->$shorthand = $args[ $name ];
	}
}
```

Doing it this way will keep the code more backward compatible - because now we use `$this->w` so changing now everything to `$this->width` would mean it would be a good thing to change it everywhere in this helper to keep the naming convention.

The only downside is the `crop => c` instead of `c => crop`. Probably all the new params will also be added in reverse way. It bothers me bit, but I still think it's a bit clearer than having two loops for normal and reveresed params. I think that a short note near w and h should be enought.

Reply:
Oh, okay. I think I missed that the properties are already called `$w` and `$h`. I think the most important thing is to make both the full name and the shorthands work. So: 👍.